docs/user/gcp/install_upi.md

```diff
@@ -0,0 +1,594 @@
+gcloud dns record-sets transaction execute --zone ${INFRA_ID}-private-zone
+```
+
+## Create firewall rules and IAM roles
+
+Copy [`03_security.py`](../../../upi/gcp/03_security.py) locally.
+
+Export variables needed by the resource definition.
+
+```sh
+export MASTER_NAT_IP=`gcloud compute addresses describe ${INFRA_ID}-master-nat-ip --region ${REGION} --format json | jq -r .address`
+export WORKER_NAT_IP=`gcloud compute addresses describe ${INFRA_ID}-worker-nat-ip --region ${REGION} --format json | jq -r .address`
+```
+
+Create a resource definition file: `03_security.yaml`
+
+```console
+$ cat <<EOF >03_security.yaml
+imports:
+- path: 03_security.py
+
+resources:
+- name: cluster-security
+  type: 03_security.py
+  properties:
+    infra_id: '${INFRA_ID}'
+    region: '${REGION}'
+
+    cluster_network: '${CLUSTER_NETWORK}'
+    network_cidr: '${NETWORK_CIDR}'
+    master_nat_ip: '${MASTER_NAT_IP}'
+    worker_nat_ip: '${WORKER_NAT_IP}'
+EOF
+```
+- `infra_id`: the infrastructure name (INFRA_ID above)
+- `region`: the region to deploy the cluster into (for example us-east1)
+- `cluster_network`: the URI to the cluster network
+- `network_cidr`: the CIDR of the vpc network (for example 10.0.0.0/16)
+- `master_nat_ip`: the ip address of the master nat (for example 34.94.100.1)
+- `worker_nat_ip`: the ip address of the worker nat (for example 34.94.200.1)
+
+Create the deployment using gcloud.
+
+```sh
+gcloud deployment-manager deployments create ${INFRA_ID}-security --config 03_security.yaml
+```
+
+The templates do not create the policy bindings due to limitations of Deployment
+Manager, so we must create them manually.
+
+```sh
+export MASTER_SA=${INFRA_ID}-m@${PROJECT_NAME}.iam.gserviceaccount.com
+gcloud projects add-iam-policy-binding ${PROJECT_NAME} --member "serviceAccount:${MASTER_SA}" --role "roles/compute.instanceAdmin"
+gcloud projects add-iam-policy-binding ${PROJECT_NAME} --member "serviceAccount:${MASTER_SA}" --role "roles/compute.networkAdmin"
+gcloud projects add-iam-policy-binding ${PROJECT_NAME} --member "serviceAccount:${MASTER_SA}" --role "roles/compute.securityAdmin"
+gcloud projects add-iam-policy-binding ${PROJECT_NAME} --member "serviceAccount:${MASTER_SA}" --role "roles/iam.serviceAccountUser"
+gcloud projects add-iam-policy-binding ${PROJECT_NAME} --member "serviceAccount:${MASTER_SA}" --role "roles/storage.admin"
+
+export WORKER_SA=${INFRA_ID}-w@${PROJECT_NAME}.iam.gserviceaccount.com
+gcloud projects add-iam-policy-binding ${PROJECT_NAME} --member "serviceAccount:${WORKER_SA}" --role "roles/compute.viewer"
+gcloud projects add-iam-policy-binding ${PROJECT_NAME} --member "serviceAccount:${WORKER_SA}" --role "roles/storage.admin"
+```
+
+Create a service account key and store it locally for later use.
+
+```sh
+gcloud iam service-accounts keys create service-account-key.json --iam-account=${MASTER_SA}
+
```

Comment:
nit: you don't need this blank line

Reply:
Those empty lines are sneaky with the back-ticks. Done.